driver/src/main/java/org/neo4j/driver/internal/pool/StandardConnectionPool.java

```diff
@@ -71,25 +65,19 @@
      */
     private final ValidationStrategy<PooledConnection> connectionValidation;
 
-    private final int connectionsPerDatabase;
     private final Clock clock;
+    private final Config config;
 
-    public StandardConnectionPool( Logging logging )
-    {
-        this( loadConnectors( logging ) );
-    }
-
-    public StandardConnectionPool( Collection<Connector> conns )
+    public StandardConnectionPool( Config config )
```

Comment:
I would actually vote for not letting Config go all the way down here - have the surface, `Driver`, ask for Config, but past that pass in component-specific arguments. So rather than asking for config here, we ask for the parameters we need (`connectionsPerDatabase` and `minIdleTimeBeforeConnectionTest`). Config would only be used by `Driver`, keeping the rest of the system from having to know about all the other config options, and making it possible to test these components without creating a config object.